sdk/azcore/core.go

```diff
@@ -109,3 +110,38 @@ func (n nopCloser) Close() error {
 func NopCloser(rs io.ReadSeeker) ReadSeekCloser {
 	return nopCloser{rs}
 }
+
+// Poller provides operations for checking the state of a long-running operation.
+// An LRO can be in either a non-terminal or terminal state.  A non-terminal state
+// indicates the LRO is still in progress.  A terminal state indicates the LRO has
+// completed successfully, failed, or was cancelled.
+type Poller interface {
```

Comment:
would it make sense to include the "Complete" Poller interface here too (e.g. this is the backing value for `GetX`, but rather for `GetXComplete`?)

Reply:
Putting the [`PollUntilDone()`](https://github.com/Azure/azure-sdk-for-go/blob/master/sdk/arm/compute/2020-09-30/armcompute/zz_generated_models.go#L63) method on the response envelope was an intentional decision.  The reasoning is to promote the "happy path" for the common case where a consumer can simply wait for the LRO to complete without needing a custom polling loop.  The other problem here is that `PollUntilDone()` returns a strongly-typed result so we can't place it inside `azcore` until we have generics (if we really wanted to, we'd either have to use an `interface{}` out param, or change pattern to be similar to track 1).

Reply:
Generally speaking, in track 2 we're taking the approach of trying to promote the easy/common APIs, leaving the "complex" APIs/patterns under a level of indirection (if that makes sense).  This is also why we don't expose the "*createRequest" and "*handleResponse" methods in the default public surface area; they're considered an advanced scenario (still looking for a clean way to solve this one).